### DISCUSSION TITLE
Assets endpoint alternative

### DISCUSSION THREAD

**Discussion by AbraamNader** [directus/directus#6994] [Feature Requests]
I've deployed a new Directus setup on AWS on a website with very high number of concurrent users.

Directus was unresponsive due to high traffic, and our content team was not able to update the content using the Admin App. (Since it was not deployed separately)

After setting this option:
`ASSETS_CACHE_TTL="7d"`
our Cloudflare caching service was able to cache the assets.
After doing so Directus was responsive again.

I also checked the PM2 monitor, and it seemed that the network requests did not even reach Directus (also CPU almost at 0%),
So my conclusion was that the assets endpoint was using too many network connections, since we used a lot of images on our frontend, multiply that by the number of users and ... all the available network resources are consumed.

Caching is only a temporary solution, so here is my suggestion:
We need an alternative to the assets endpoint my idea was to access the assets storage directly (in my case, make my S3 bucket public).

However I was not able to do so since the naming of files was seemed unpredictable, sometimes there is an extra dot, sometimes there wasn't,
so reverse engineering the path was a hard task.

so I hope the documentation had a pattern that we can use in order to get the path with the right size/quality.

Any other suggestions are appreciated as well.

Thanks

**Comment by rijkvanzanten** [comment]
> So my conclusion was that the assets endpoint was using too many network connections

How are you running Directus itself? What are you using as a reverse proxy? Repeated requests to pre-generated assets should stream the file directly from the storage adapter, which really shouldn't affect the server too much compared to "regular" requests for data 🤔 Could it be that the bottleneck is the webserver that processes the requests to/from directus itself?

**Reply by AbraamNader** [reply]
I used nginx to pass the requests to Directus, just like the documentation suggested.
The Directus app is currently stable, but if for any reason cloudflare cache was purged, it would probably become unresponsive again.

The bottleneck here is the maximum number of network request the server can handle,
A typical user calls one API per view in our frontend
but since Directus serves the static files as well this adds like a hundred more API requests per user per view.

This is why I needed to offload these connections directly to S3, since they are just static resources (they are generated on upload) and they need not be processed by Directus, since they are public anyway.

In the future, We will use Docker to scale it better, but we needed to know how much a single Directus instance can take, to plan accordingly, currently I am thinking about having a separate instance for the Admin App, and scale the API instances (Reads) as traffic goes up or down.

**Reply by rijkvanzanten** [reply]
Yeah, makes sense! Besides using the S3 path directly, I'm wondering what we can do to optimize those static asset requests some more, as even though the permissions are checked (if applicable), that should still be fairly light in terms of traffic

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
